### PR TITLE
janitor: bump rust-version in Cargo.toml

### PR DESCRIPTION
**Pull request by tronical**
* 1.64 for compiler and winit due to the ffi types
* 1.62 for the interpreter and core for the enum defaults

### CODE REVIEW

**Comment by tronical** [comment]
Is this correct?

File: internal/core/Cargo.toml [hunk 1]
```diff
@@ -12,7 +12,7 @@ repository = "https://github.com/slint-ui/slint"
 homepage = "https://slint-ui.com"
 categories = ["gui", "development-tools"]
 keywords = ["gui", "ui", "toolkit", "graphics", "design"]
-rust-version = "1.60"
+rust-version = "1.62"
```
Comment (ogoffart) [inline internal/core/Cargo.toml:15]: do we have a test for this?

File: internal/interpreter/Cargo.toml [hunk 2]
```diff
@@ -12,7 +12,7 @@ repository = "https://github.com/slint-ui/slint"
 homepage = "https://slint-ui.com"
 categories = ["gui", "rendering::engine"]
 keywords = ["gui", "toolkit", "graphics", "design", "ui"]
-rust-version = "1.60"
+rust-version = "1.62"
```
[the remark this answers is not in the document]
Comment (tronical) [reply internal/interpreter/Cargo.toml:15]: Ok, thanks :)

File: internal/core/Cargo.toml (hunk 1 above)
Comment (tronical) [reply internal/core/Cargo.toml:15]: No, the CI only tests with the "global" MSRV 1.64.

**Comment by ogoffart** [comment]
Given that the winit backend needs 64, we can as well mandate 64 for everything as there is no practical adventage not to. Is there?

And if we support rust 1.64,  we can use the workspace.package stuff https://doc.rust-lang.org/nightly/cargo/reference/workspaces.html#the-package-table

**Comment by tronical** [comment]
> Given that the winit backend needs 64, we can as well mandate 64 for everything as there is no practical adventage not to. Is there?

I agree. I vaguely recall this being discussed differently in the past, but I totally agree that it's easiest for us to maintain just one global number.

> And if we support rust 1.64, we can use the workspace.package stuff https://doc.rust-lang.org/nightly/cargo/reference/workspaces.html#the-package-table

That's even better. I hope that's copied into `Cargo.toml` when uploading to crates.io.

**Comment by ogoffart** [comment]
> I vaguely recall this being discussed differently in the past, but I totally agree that it's easiest for us to maintain just one global number.

I think we said we can have the tools such as the lsp or the viewer use higher version if needed. Or maybe even a difference between the "runtime" and the build-time crates.
But in this case, since the winit backend needs the 64, there is not much of a case.

**Review by ogoffart — APPROVED** [review]

**Pull request merged**